langchain/llms/custom_endpoint.py

```diff
@@ -0,0 +1,133 @@
+"""Wrapper around custom APIs."""
+from typing import Any, Dict, List, Mapping, Optional
+
+import requests
+import asyncio
+from functools import partial
+from pydantic import Extra, root_validator
+
+from langchain.llms.base import LLM
+from langchain.llms.utils import enforce_stop_tokens
+from langchain.utils import get_from_dict_or_env
+
+VALID_TASKS = ("text2text-generation", "text-generation")
+
+
+class CustomEndpoint(LLM):
+    """Wrapper around custom Inference Endpoints, related to HuggingFaceHub Inference Endpoints.
+
+    Only supports `text-generation` and `text2text-generation` for now.
+
+    Example:
+        .. code-block:: python
+
+            from langchain.llms import CustomEndpoint
+            endpoint_url = "https://api/endpoint/
+            llm = CustomEndpoint(
+                endpoint_url=endpoint_url,
+                api_key="my-api-key"
+            )
+    """
+
+    endpoint_url: str = ""
+    """Endpoint URL to use."""
+    task: Optional[str] = None
```

Comment:
This isn't actually `Optional`, since we are raising an error when called below. It should be validated with a `@validator` or `@root_validator`